backend/src/api/base.clj

```diff
@@ -13,38 +13,44 @@
   (player-view/get-game-as-player (persistence/fetch-game game-id) player-id))
 
 (defn play-card-as-player
-  [game-id player index row-id & target]
+  [game-id player-id card-id row-id & target]
   (let [game-state (persistence/fetch-game game-id)]
-      (if (= (:status (get-game game-id player)) messages/play)
+      (if (= (:status (get-game game-id player-id)) messages/play)
           (do
-            (persistence/save-game (play-card/play-card game-state (conversions/player-num game-state player) index row-id (first target)))
-            (get-game game-id player))
+            (persistence/save-game (play-card/play-card game-state player-id card-id row-id (first target)))
+            (get-game game-id player-id))
           {:error messages/out-of-turn})))
 
 (defn ^:private create-empty-game
-  "Creates a new instance of a game"
+  "Creates a new instance of a game lobby"
   [ini-config]
   (let [game-id (persistence/next-id)]
     (persistence/save-game
-      (assoc (create-game/new-game ini-config)
-             :game-id game-id))
-    game-id))
+      {:status messages/no-opp
+       :game-id game-id
+       :player-ids [(generators/player-uuid {})]})))
 
 (defn add-player
   "Adds a player to a game"
   [game-id]
-  (let [saved-game (persistence/fetch-game game-id)
-        players-connected (count (:player-ids saved-game))]
-    (if (> players-connected 1)
-      {:error messages/too-many-players}
-      (let [game-state (persistence/save-game
-                         (generators/player-uuid
-                           saved-game))]
-        (get-game game-id (last (:player-ids game-state)))))))
+  (let [lobby (persistence/fetch-game game-id)
+        players-connected (count (:player-ids lobby))]
+    (cond
+      (> players-connected 1) {:error messages/too-many-players}
+      (< players-connected 1) {:error messages/lobby-not-created}
+      :else (let [second-uuid (generators/player-uuid lobby)]
+              (-> (create-game/new-game
+                    {:player-ids [(first (:player-ids lobby))
```

Comment:
Almost went crazy during the weekend til I realized that I needed to pass the player-ids to the `new-game` constructor (which is *perfectly logical, I might add*) instead of putting them by `assoc` post-construction as with the `game-id`. In fact, I think the `game-id`should also be passed in and added to the game in `new-game` to ease the logic (everything is done at construction instead of modifying stuff afterwards)